### DISCUSSION TITLE
Interrupting handshake on QUIC_CONNECTION_EVENT_PEER_CERTIFICATE_RECEIVED

### DISCUSSION THREAD

**Discussion by fernandoherrerav** [microsoft/msquic#2906] [Q&A] [answered]
Hi, I am validating some certificate conditions on QUIC_CONNECTION_EVENT_PEER_CERTIFICATE_RECEIVED on the server connection callback. If they fail, How can I gracefully prevent the handshake from succeeding?

**Comment by nibanks** [comment, marked as answer]
You can return an error. See the code [here](https://github.com/microsoft/msquic/blob/main/src/core/connection.c#L3052) that handles the return code from your event.

**Reply by fernandoherrerav** [reply]
Thank you
